### DISCUSSION TITLE
add prometheus link in query editor back

### DISCUSSION THREAD

**Discussion by juliantaylor** [grafana/grafana#49382] [General]
The prometheus link in the query editor was incredibly useful to tune your queries in the prometheus UI with all the template variables expanded.
It was removed in 8.5 see https://github.com/grafana/grafana/issues/46944 (the issue itself is easily worked around).
I used it on a regular basis and it made working with grafana much more convinient by being able to combine the features of grafana and prometheus when building queries.
It would be great to have it back

**Comment by pjjw** [comment]
+1 here

**Comment by bitzhuxb** [comment]
It will be appreciated if the "prometheus" button can be setting in grafana.ini

**Comment by itsmylife** [comment]
Thank you for submitting a feature request. We’ve reviewed the request, other possible implementation options and have decided not to implement this feature for the following reasons:
- Different Prometheus setups don’t have the web ui available
- Supporting all different scenarios is cumbersome

Even when we make the decision not to pursue a feature, the submission encourages conversations and ideas within our development team to continue making improvements, so please keep sharing! If you think this is something Grafana should have feel free to send PRs.

**Reply by pjjw** [reply]
not all support it, but that's why it's gated behind a user-controlled setting. don't understand that reasoning.

**Comment by juliantaylor** [comment]
how about a button that gives you the evaluated expression to copy paste into your prometheus then?

**Reply by juliantaylor** [reply]
having another look the current gui this exists now in the query inspector, I this is cumbersome but acceptable
